ios/brave-ios/Sources/Brave/Frontend/Browser/Search/BraveSearchResultAdManager.swift

```diff
@@ -15,9 +15,9 @@ class BraveSearchResultAdManager: NSObject {
 
   private var searchResultAds = [String: BraveAds.SearchResultAdInfo]()
 
-  init?(url: URL, rewards: BraveRewards, isPrivateBrowsing: Bool) {
-    if !BraveAds.shouldSupportSearchResultAds() || !BraveSearchManager.isValidURL(url)
-      || isPrivateBrowsing
+  init?(url: URL, rewards: BraveRewards, isPrivateBrowsing: Bool, isAggressiveAdsBlocking: Bool) {
+    if isPrivateBrowsing || isAggressiveAdsBlocking || !BraveSearchManager.isValidURL(url)
+      || !BraveAds.shouldSupportSearchResultAds()
```

Comment:
Maybe should just pass the domain over and isolate all this business logic in this place?

Reply:
Unfortunately, this requires to add @MainActor attribute to classes `BraveSearchResultAdManager` and `BraveSearchResultAdScriptHandler` , which in turn will require changing TabContentScript protocol and all it's implementations. This refactoring is beyond current PR and it was agreed in DM that it should be done separately.